.cursor/rules/solidity-styles.mdc

```diff
@@ -0,0 +1,69 @@
+---
+description:
+globs: *.sol
+alwaysApply: false
+---
+# Optimism Solidity Style Guide
+
+Applies to Solidity files.
+
+## Comments
+- Use triple-slash solidity natspec comment style
+- Always use `@notice` instead of `@dev`
+- Use a line-length of 100 characters
+- Custom tags:
+  - `@custom:proxied`: Add to a contract whenever it's meant to live behind a proxy
+  - `@custom:upgradeable`: Add to a contract whenever it's meant to be inherited by an upgradeable contract
+  - `@custom:semver`: Add to `version` variable which indicate the contracts semver
+  - `@custom:legacy`: Add to an event or function when it only exists for legacy support
+  - `@custom:network-specific`: Add to state variables which vary between OP Chains
+
+## Errors
+- Use `require` statements for simple assertions
+- Use `revert(string)` if throwing an error where an assertion is not being made
+- Error strings must have format `"{ContractName}: {message}"` where `message` is lowercase
```

Comment:
@smartcontracts This seems out of date, we use custom errors in source code and `require` is allowed in tests/scripts but must have a string arg